pkg/gitconfig/configs.go

```diff
@@ -0,0 +1,358 @@
+	for _, cfg := range []*Config{
+		c.env,
+		c.worktree,
+		c.local,
+		c.global,
+		c.system,
+		c.Preset,
+	} {
+		if cfg == nil || cfg.vars == nil {
+			continue
+		}
+		if v, found := cfg.vars[key]; found {
+			return v
+		}
+	}
+
+	debug.Log("no value for %s found", key)
+
+	return ""
+}
+
+// GetGlobal specifically ask the per-user (global) config for a key.
+func (c *Configs) GetGlobal(key string) string {
+	if c.global == nil {
+		return ""
+	}
+
+	if v, found := c.global.vars[key]; found {
+		return v
+	}
+
+	debug.Log("no value for %s found", key)
+
+	return ""
+}
+
+// GetLocal specifically asks the per-directory (local) config for a key.
+func (c *Configs) GetLocal(key string) string {
+	if c.local == nil {
+		return ""
+	}
+
+	if v, found := c.local.vars[key]; found {
+		return v
+	}
+
+	debug.Log("no value for %s found", key)
+
+	return ""
+}
+
+// IsSet returns true if this key is set in any of our configs.
+func (c *Configs) IsSet(key string) bool {
+	for _, cfg := range []*Config{
+		c.env,
+		c.worktree,
+		c.local,
+		c.global,
+		c.system,
+		c.Preset,
+	} {
+		if cfg.IsSet(key) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// SetLocal sets (or adds) a key only in the per-directory (local) config.
+func (c *Configs) SetLocal(key, value string) error {
+	if c.local == nil {
+		if c.workdir == "" {
+			return fmt.Errorf("no workdir set")
+		}
+		c.local = &Config{}
+	}
+
+	return c.local.Set(key, value)
+}
+
+// SetGlobal sets (or adds) a key only in the per-user (global) config.
+func (c *Configs) SetGlobal(key, value string) error {
+	if c.global == nil {
+		c.global = &Config{}
+	}
+
+	return c.global.Set(key, value)
+}
+
+// SetEnv sets (or adds) a key in the per-process (env) config. Useful
+// for persisting flags during the invocation.
+func (c *Configs) SetEnv(key, value string) error {
+	if c.env == nil {
+		c.env = &Config{
+			noWrites: true,
+		}
+	}
+
+	return c.env.Set(key, value)
+}
+
+// UnsetLocal deletes a key from the local config.
+func (c *Configs) UnsetLocal(key string) error {
+	if c.local == nil {
+		return nil
+	}
+
+	return c.local.Unset(key)
+}
+
+// UnsetGlobal delets a key from the global config.
+func (c *Configs) UnsetGlobal(key string) error {
+	if c.global == nil {
+		return nil
+	}
+
+	return c.global.Unset(key)
+}
+
+// Keys returns a list of all keys from all available scopes. Every key has section and possibly
+// a subsection. They are seprated by dots. The subsection itself may contain dots. The final
+// key name and the section MUST NOT contain dots.
+//
+// Examples
+//   - remote.gist.gopass.pw.path -> section: remote, subsection: gist.gopass.pw, key: path
+//   - core.timeout -> section: core, key: timeout
+func (c *Configs) Keys() []string {
+	keys := make([]string, 0, 128)
+
+	for _, cfg := range []*Config{
+		c.Preset,
+		c.system,
+		c.global,
+		c.local,
+		c.worktree,
+		c.env,
+	} {
+		if cfg == nil {
+			continue
+		}
+		if cfg.vars == nil {
+			continue
+		}
```

Comment:
```suggestion
		if cfg == nil || cfg.vars == nil {
			continue
		}
```

Reply:
Actually we don't even have to check for `cfg.vars == nil`. range handles that properly.